### PR TITLE
Clear cache

### PR DESCRIPTION
**Pull request by jordisala1991**
## Subject

I am targeting this branch, because {reason}.

Closes #{put_issue_number_here}.

## Changelog

```markdown
### Added
- Added some `Class::newMethod()` to do great stuff.

### Changed

### Deprecated

### Removed

### Fixed

### Security
```

### CODE REVIEW

File: phpunit.xml.dist [hunk 1]
```diff
@@ -37,5 +37,8 @@ It's auto-generated by sonata-project/dev-kit package.
     <php>
         <ini name="precision" value="8" />
         <env name="SYMFONY_DEPRECATIONS_HELPER" value="max[self]=0" />
+        <env name="KERNEL_CLASS" value="\Sonata\PageBundle\Tests\App\AppKernel" />
+        <server name="APP_ENV" value="test" force="true" />
+        <server name="APP_DEBUG" value="false" />
```
Comment (VincentLanglet) [inline phpunit.xml.dist:42]: This file is auto generated ; do you have something in mind ?

**Comment by jordisala1991** [comment]
I still need to understand why coverage does not increase, it does not make any sense tbh. there is some files on dependency injection (the extension class) that somehow is not covered.

**Comment by jordisala1991** [comment]
Still, this would improve by a LOT the DX on Sonata. And it would make all test run independent from changes on config and all (previously I had to manually edit the cache folder to get a clean build)

File: phpunit.xml.dist (hunk 1 above)
Comment (jordisala1991) [reply phpunit.xml.dist:42]: Yep, the same we generate the console file, we can generate the namespace for the AppKernel file.

**Comment by jordisala1991** [comment]
There are 3 things here:
1. use variables to define kernel class like in normal sf projects (and avoid extra code to define on each test case which kernel to use)
2. clear cache to avoid broken test when doing changes to code.
3. if possible, see why the coverage does not increase.

File: phpunit.xml.dist [hunk 2]
```diff
@@ -37,5 +37,8 @@ It's auto-generated by sonata-project/dev-kit package.
     <php>
         <ini name="precision" value="8" />
         <env name="SYMFONY_DEPRECATIONS_HELPER" value="max[self]=0" />
+        <env name="KERNEL_CLASS" value="\Sonata\PageBundle\Tests\App\AppKernel" />
```
Comment (eerison) [inline phpunit.xml.dist:40]: hey @jordisala1991  I tried to do this before But @VincentLanglet mentioned  that this file is generated automatically

Comment (VincentLanglet) [reply phpunit.xml.dist:40]: He plans to update the dev-kit file generator

**Comment by jordisala1991** [comment]
Now it reflects the real coverage, and we dont have issues about regenerating cache. wdyt @VincentLanglet ?

**Review by VincentLanglet — COMMENTED** [review]
Seems nice. We will have to update all the projects ?

**Comment by jordisala1991** [comment]
I think most of them yes, dev kit will handle part of the update and we will have to do the rest.

**Comment by jordisala1991** [comment]
This will be dev kit PR: https://github.com/sonata-project/dev-kit/pull/2154

Keep in mind that other projects might not benefit from increase coverage, but all of them should get the other benefits (remove the getkernelclass funct and avoid problems with cache)

**Comment by eerison** [comment]
why was it closed 👀

**Comment by jordisala1991** [comment]
Because it is already done (not merged on all branches, but on the stable ones. I will merge this one manually since it is probably the one which will cause more conflicts.

**Pull request closed without merging**